### PR TITLE
Add purchase & balance commands

### LINKED ISSUE #2: 💰 Bounty 2: Discord Bot - Purchase & Balance Commands - $60 (1000 $RLY) 💰
=========================

**Bounty 2: Discord Bot - Purchase & Balance Commands**

**Status: 🟢 ACTIVE 🟢**

**Reward: 💰 $60 (paid via 1000 $RLY) 💰**

**Time Estimate: 1 Hour** 

=========================

**Bounty Overview:**

A list of Creator Coin commands that help bot admins add information to their communities (described below) to be added to the current rallyrolebot made by Calvin. These commands should be built to be consistent with the current design and be submitted to the Rally community repository as a PR to be reviewed. Once approved, they will be merged and the Bounty will be 
awarded.

**Repository link here: https://github.com/CreatorCoinTools/RallyRoleBot**

**To Discuss or Claim this bounty, please join our Discord: https://discord.gg/aPCQYQDQzp**

**Docs:** https://developer.rally.io/#/

**Minimum Requirements:**

- Must use the Rally.io App Data API
- Implement the commands as described below: 

**- $setpurchasemsg** <message> - Allow Admin/Botmasters to change the $purchase message.

**- $purchase -** Display this default message:

“You can purchase <CoinName> at <CreatorsRallyURL> by using a Credit/Debit card or a number of different Crypto Currencies!  Buying <CoinName> earns rewards, supports the community, and you can even get VIP Status!  (hint: there’s a secret VIP room for users who hold over X # of <CoinName> ;)

**- $donate** - Display message:

“You can donate <CoinName> to <CREATORNAME> by going to <CreatorRallyPage> - Your donation helps grow and support the community and creator - Plus, there are 10 tiers of Donation badges to earn to show off your support! 

**- $setdonatemsg <message>** - Allow Admin/Botmasters to change the $donate message.

**- $balance <username>** - Display Users Coin, if they’ve registered their rally network ID with the bot. (<username> = discord username). If <username> hasn’t linked, the response should be “@<Username> hasn’t verified their wallet yet! Type !join

### PR DESCRIPTION
**Pull request by Echolon166**
Closes #2 

Added all 5 commands which was requested in the bounty.

### CODE REVIEW

File: rallyrolebot/cogs/rally_cog.py [hunk 1]
```diff
@@ -32,3 +32,17 @@ async def cog_command_error(self, ctx, error):
     @commands.dm_only()
     async def set_rally_id(self, ctx, rally_id):
         data.add_discord_rally_mapping(ctx.author.id, rally_id)
+
+    @commands.command(name="balance", help="View your balance")
+    @validation.is_wallet_verified()
+    async def balance(self, ctx):
+        rally_id = data.get_rally_id(ctx.message.author.id)
+        balances = rally_api.get_balances(rally_id)
+        await ctx.send(
```
Comment (pashneal) [inline rallyrolebot/cogs/rally_cog.py]: Use utils.pretty_print to preserve a common theme among RallyRoleBot outputs.

File: rallyrolebot/cogs/channel_cog.py [hunk 2]
```diff
@@ -109,3 +109,29 @@ async def get_channel_mappings(self, ctx):
                 ]
             )
         )
+
+    @commands.command(name="set_purchase_message", help="Change the $purchase message")
+    @validation.owner_or_permissions(administrator=True)
+    async def set_purchase_message(self, ctx, message):
+        data.set_purchase_message(ctx.guild.id, ctx.message.content.split(' ', 1)[1])
```
Comment (pashneal) [inline rallyrolebot/cogs/channel_cog.py]: Use discord.py keyword arguments for a cleaner function
```suggestion
    async def set_purchase_message(self, ctx, *, message):
        data.set_purchase_message(ctx.guild.id, message)
```

File: rallyrolebot/cogs/channel_cog.py [hunk 3]
```diff
@@ -109,3 +109,29 @@ async def get_channel_mappings(self, ctx):
...
+        message = data.get_purchase_message(ctx.guild.id)
+        if message is not None:
+            await ctx.send(message)
+        else:
+            await ctx.send(DEFAULT_PURCHASE_MESSAGE)
+
+    @commands.command(name="set_donate_message", help="Change the $donate message")
+    @validation.owner_or_permissions(administrator=True)
+    async def set_donate_message(self, ctx, message):
+        data.set_donate_message(ctx.guild.id, ctx.message.content.split(' ', 1)[1])
```
Comment (pashneal) [inline rallyrolebot/cogs/channel_cog.py]: Use discord.py keyword arguments for a cleaner function
```suggestion
    async def set_donate_message(self, ctx, *, message):
        data.set_donate_message(ctx.guild.id, message)
```

File: rallyrolebot/validation.py [hunk 4]
```diff
@@ -69,3 +70,12 @@ async def is_valid_coin(ctx, coin_name):
         await ctx.send("Coin " + coin_name + " does not seem to exist")
         return False
     return True
+
+def is_wallet_verified():
+    async def extended_check(ctx):
+        rally_id = data.get_rally_id(ctx.message.author.id)
+        if rally_id is None:
+            raise errors.WalletNotVerified(ctx.message.author.mention + " hasn’t verified their wallet yet! Type !join")
+        return rally_id
```
Comment (pashneal) [inline rallyrolebot/validation.py]: No need to return rally_id, return True for readibility

File: rallyrolebot/cogs/rally_cog.py (hunk 1 above)
Comment (Echolon166) [reply rallyrolebot/cogs/rally_cog.py]: Done.

File: rallyrolebot/cogs/channel_cog.py (hunk 2 above)
Comment (Echolon166) [reply rallyrolebot/cogs/channel_cog.py]: Done.

File: rallyrolebot/cogs/channel_cog.py (hunk 3 above)
Comment (Echolon166) [reply rallyrolebot/cogs/channel_cog.py]: Done.

File: rallyrolebot/validation.py (hunk 4 above)
Comment (Echolon166) [reply rallyrolebot/validation.py]: Done.

**Review by pashneal — APPROVED** [review]
Approved for #2

**Pull request merged**